### DISCUSSION TITLE
테스트 실행 오류

### DISCUSSION THREAD

**Discussion by LeeJongbokz** [naver/ngrinder#923] [Q&A]
안녕하세요!
Ngrinder 테스트 실행 오류와 관련된 오류에 대해서 질문 드리고자 글 남깁니다 :)
시간 괜찮으실 때 답변 주시면 매우 감사드리겠습니다!

아래와 같은 에러가 떠서 구글링을 해봤는데
![image](https://user-images.githubusercontent.com/8718430/196772351-5707a505-2d87-4594-adc1-7b2695bf7a94.png)

다음과 같은 답변을 발견하였고, 제 현재 JDK 버전을 확인해보니
![image](https://user-images.githubusercontent.com/8718430/196772605-bc2f93c6-3ec3-475e-9238-e7b714b08ebe.png)

다음과 같음을 확인하였습니다.
![image](https://user-images.githubusercontent.com/8718430/196772663-c263da89-3a6d-48f3-9f08-70045135cf1c.png)

- 여기서 궁금한 점은 제 로컬의 JDK 버전과 해당 에러가 어떤 관계에 있는건지를 여쭙고자 
  이 글을 작성하게 되었습니다!
  편하실 때 답변 주시면 매우 감사드리겠습니다!
  감사합니다~

**Comment by donggyu04** [comment]
@LeeJongbokz 

https://github.com/naver/ngrinder/blob/develop/ngrinder-runtime/src/main/java/org/ngrinder/dns/NameServiceProxy.java#L59
위 코드 라인에서 `ClassNotFoundException` 이 발생되는 것 같습니다.

아마도 java18 에서 위 코드의 try 블록 내부에서 오류가 발생된 것 같은데 그 부분 확인해 보시면 될 것 같습니다.

> java.net.InetAddress$NameService 이 클래스가 java18 에 없지 않을까 싶습니다.

**Reply by LeeJongbokz** [reply]
답변 감사드립니다~

**Comment by junoyoon** [comment]
저희는 /etc/hosts 없이  사용자가 입력한 ip/host 쌍으로 ip resolve 를 하기 위해, JDK internal 의 기능을 사용합니다.
이 부분이 jdk18 에서 제거된 것으로 보입니다.

@imbyungjun 님 이슈 등록해 주시고, 해결 방법 한번 찾아봐 주실래요?

**Reply by LeeJongbokz** [reply]
답변 감사합니다!

**Comment by donggyu04** [comment]
@junoyoon @imbyungjun 

java18 코드를 좀 보니 기존 NameService 기능을 하는게 java18 부터  [InetAddressResolver](https://docs.oracle.com/en/java/javase/18/docs/api/java.base/java/net/spi/InetAddressResolver.html) 로 대체된 것 같습니다.
[InetAddressResolverProvider](https://docs.oracle.com/en/java/javase/18/docs/api/java.base/java/net/spi/InetAddressResolverProvider.html) 를 구햔하여 우리가 커스터 마이징한 `InetAddressResolver` 구현체가 리턴되게 하면 시스템 전체적으로 커스터 마이징한 `InetAddressResolver`가 사용되어 동작되는 것 같습니다. (`InetAddressResolverProvider` 를 따로 제공하지 않으면 내부 빌트 인 InetAddressResolver 가 사용 됨)

**Reply by LeeJongbokz** [reply]
답변 감사드립니다~
